code/game/objects/items.dm

```diff
@@ -1124,6 +1125,43 @@
 	animate(time = 1)
 	animate(alpha = 0, time = 3, easing = CIRCULAR_EASING|EASE_OUT)
 
+/obj/item/proc/handle_dissolve(mob/user)
+	if(!HAS_TRAIT(user, TRAIT_YAUTJA_TECH))
+		to_chat(user, SPAN_WARNING("You have no idea what this even does..."))
+		return FALSE
+	if(user.alpha < 255)
+		to_chat(user, SPAN_BOLDWARNING("It would not be safe to attempt this while cloaked!"))
+		return FALSE
+	if(anchored)
+		to_chat(user, SPAN_WARNING("\The [src] cannot be moved by any means, why dissolve it?"))
+		return FALSE
+
+	var/mob/living/location = loc
+	if(istype(location))
+		to_chat(user, SPAN_WARNING("You cannot dissolve this while it is being held!"))
+		return FALSE
+
+	dissolve(user)
+	return TRUE
+
+/obj/item/proc/dissolve(mob/user)
```

Comment:
Move this logic to be under `/obj/item/tool/yautja_cleaner`, it should look something like
```
if(!target.handle_dissolve())
    return

dissolve(target)
```